### PR TITLE
Release fixes

### PR DESCRIPTION
**Pull request by willespencer**
### Summary 

This pull request fixes a number of small bugs that have been built into CoursePlan sometime between the start of the project and now.

- [x] Update CoursePlan logo (rounded corners) and ensure that our site is always referred to as "CoursePlan"
- [x] Reverse the order of the semesters (now from future to past)
- [x] Re-enable multiple majors on the site
- [x] Factor out 404 and landing page code to other files
- [x] Replace css dots used for course cards and settings menus with imgs 
- [x] Change edit color/credits menu to better handle when they must expand on the left side
- [ ] Come up with a new edit color algorithm to handle multiple breakpoints (left commented out)
- [x] Prevent duplicate semesters by disabling buttons for both adding and editing semesters
- [x] Make it much easier to drag elements to semesters, especially for empty semesters and over the '+COURSE' button, without courses ever overlapping with that button.
- [x] Remove empty space between the '+COURSE' button and the list of courses if adjacent semesters have a lot of courses.

### Test Plan 

Ensure editing colors/credits menus display as expected for every 2nd semester (on the regular display) and every 4th semester (on the compact display).

image

Test both adding and editing semesters have a disabled button when trying to add/edit a duplicate (with the exception of editing a semester to it's current season/year), and that semesters are always kept in the new order.

image

Test dragging on both compact and regular views to verify that it is easier to drag courses to empty and other semesters. Ensure that the +COURSE button is never covered by a course and that there is space between courses and that button.

image

image

### Notes 

Just like it did when it was first updated, the edit colors menu does not work well on breakpoints where there are not exactly 4 semesters (for compact) or 2 semesters (for regular) on a line, because the code changes the functionality based on the semester index. This should be updated in a later pull reuqest, potentially with the incomplete method I left in the codebase with a TODO.

### CODE REVIEW

File: src/assets/images/404.svg [hunk 1]
```diff
@@ -1,21 +1,14 @@
...
-</g>
-<defs>
-<clipPath id="clip0">
-<rect width="459.673" height="143.159" fill="white"/>
-</clipPath>
-</defs>
+<svg width="440" height="139" viewBox="0 0 440 139" fill="none" xmlns="http://www.w3.org/2000/svg">
+<path fill-rule="evenodd" clip-rule="evenodd" d="M23.511 85.3361L7.51096 85.3361L39.4461 15.9895C40.2626 14.2167 42.0359 13.081 43.9877 13.081L69.7339 13.081L37.1447 85.3566C30.011 85.3361 25.511 85.3361 23.511 85.3361Z" fill="#92C3E6"/>
+<path d="M12.5215 113C9.76009 113 7.52151 110.761 7.52151 108L7.52151 88H111.522C114.283 88 116.522 90.2386 116.522 93V108C116.522 110.761 114.283 113 111.522 113H12.5215Z" fill="#508197"/>
+<path d="M100.522 116V134C100.522 136.761 98.2829 139 95.5215 139H77.5215C74.7601 139 72.5215 136.761 72.5215 134V116H100.522Z" fill="#105351"/>
```
Comment (hanywang2) [inline src/assets/images/404.svg:4]: really nice fix. Great logic and implementation

File: src/components/404.vue [hunk 2]
```diff
@@ -1,14 +1,7 @@
...
-    <div class="container p-0">
-        <div class="row top-bar justify-content-between no-gutters">
-            <div class="col-10">
-                <img class="logo" src="@/assets/images/logo.svg" alt = "logo" />
-              </div>
-        </div>
-    </div>
+    <topbar />
     <div class = "message-container">
       <img class = "img-404" src="@/assets/images/404.svg" alt = "404">
```
Comment (hanywang2) [inline src/components/404.vue:6]: Absolutely beautiful change with 404 page

File: src/components/Login.vue [hunk 3]
```diff
@@ -83,7 +77,7 @@
             <div class="row semester phonepad m-0">
                 <div class = "col-12 col-md-5 comment" >
                     <h1 class= "head">Plan Your Semesters</h1>
-                    <p class= "sub">Use Courseplan’s semesterly planner to choose courses well in advance and ensure that you never miss a requirement</p>
+                    <p class= "sub">Use CoursePlan’s semesterly planner to choose courses well in advance and ensure that you never miss a requirement</p>
```
Comment (hanywang2) [inline src/components/Login.vue:80]: Thank you

File: src/components/Footer.vue [hunk 4]
```diff
@@ -0,0 +1,59 @@
+<template>
+  <div class="container-fluid footer p-0 page_404">
+    <div class= "row footer justify-content-center m-0">
+      <div class="col-3 footer">
+        <a href ="https://www.cornelldti.org" class= "footer"> Cornell DTI @ {{getYear()}}</a>
+      </div>
+      <div class="col-3">
+        <p class= "footer">Built with LOVE</p>
```
Comment (hanywang2) [inline src/components/Footer.vue:8]: Is there a reason we're not using the heart emoji anymore?

File: src/assets/images/404.svg (hunk 1 above)
Comment (hanywang2) [reply src/assets/images/404.svg:4]: ^^^ this is a joke

File: src/components/Footer.vue (hunk 4 above)
Comment (willespencer) [reply src/components/Footer.vue:8]: I looked at the designs and realized they didn't have it - don't know if they ever did or it got updated

File: src/assets/images/404.svg (hunk 1 above)
Comment (willespencer) [reply src/assets/images/404.svg:4]: <3

**Pull request merged**